client/js/lounge.js

```diff
@@ -330,14 +330,50 @@ $(function() {
 
 	const getCloseDisplay = (target) => {
 		if (target.hasClass("lobby")) {
-			return "Disconnect";
+			return "Remove";
 		} else if (target.hasClass("channel")) {
 			return "Leave";
 		}
 
 		return "Close";
 	};
 
+	const isConnected = (target) => !target.parent().hasClass("not-connected");
+
+	const getToggleConnectionDisplay = (target) => {
+		if (isConnected(target)) {
+			return "Disconnect";
+		}
+
+		return "Connect";
+	};
+
+	let clickedNetwork;
+	contextMenuFactory.addContextMenuItem({
```

Comment:
@xPaw raised a good point, could you split this into 2 items? I believe you'd have your `isConnected(target)` in `check`. That would simplify things like `className` 😅

Reply:
Also move all this code into context menu factory.

Reply:
That complicates things like `isConnected()`, will need to play with this